eq-author-api/src/validation/index.js

```diff
@@ -27,14 +36,18 @@ module.exports = questionnaire => {
 
       const fieldname = dataPath.pop();
 
-      const objectType = dataPath[dataPath.length - 2];
+      let objectType = dataPath[dataPath.length - 1];
+      if (!isNaN(objectType)) {
+        // Must be in array of object type so get object type
+        // e.g. /sections/0/pages/0/answers/0/options/0/label
+        objectType = dataPath[dataPath.length - 2];
+      }
 
       const contextPath = dataPath.slice(1).join(".");
       const contextObj = get(questionnaire, contextPath);
-
       validationErrors.push({
         id: contextObj.id,
```

Comment:
I presume this is the id of the multiple choice answer?

Reply:
No, its the ID of the object the invalid field is in. If option label is invalid then it's ID of option. If additionalAnswer.label is invalid then it's id of additionalAnswer Textfield object.